### PR TITLE
Auto Generate Plain Text Content from HTML

### LINKED ISSUE #437: Auto Generate Plain Text Content from HTML
We require both HTML and Plain Text content for improved deliverability. In some cases, only HTML may be available. We need a function that automatically converts a given HTML string to the Plain Text equivalent.

### PR DESCRIPTION
**Pull request by cmckni3**
Add plain text converter using xpath and libxml2

Close #437

### CODE REVIEW

**Comment by SendGridDX** [comment]
[CLA assistant check](https://cla.sendgrid.com/sendgrid/sendgrid-php?pullRequest=555) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.sendgrid.com/sendgrid/sendgrid-php?pullRequest=555) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.sendgrid.com/check/sendgrid/sendgrid-php?pullRequest=555) it.</sub>

**Comment by cmckni3** [comment]
Working on adding some tests

**Comment by cmckni3** [comment]
@thinkingserious Excellent, I still have some work to do. The xpath query doesn't work for deeply nested content. I appreciate the swag!

**Comment by thinkingserious** [comment]
No worries @cmckni3. Thanks for taking your time to contribute!

**Comment by jopanel** [comment]
Hi,

Would like to consider adding a regex version of this to your class. Referenced [here](https://github.com/sendgrid/sendgrid-php/pull/462). 

Thanks

File: lib/helpers/mail/Mail.php [hunk 1]
```diff
@@ -758,6 +805,9 @@ public function getType()
     public function setValue($value)
     {
         $this->value = mb_convert_encoding($value, 'UTF-8', 'UTF-8');
+        if ($this->type === 'text/plain') {
```
Comment (thinkingserious) [inline lib/helpers/mail/Mail.php]: I don't think we should call this here, instead I propose we document this function in USE_CASES.md and the developer can optionally use this functionality when desired. That way we don't introduce new unexpected behavior for current users.

**Comment by thinkingserious** [comment]
Hi @cmckni3,

Are you still interested in completing this one? No worries, if not.

With Best Regards,

Elmer

**Comment by cmckni3** [comment]
Apologies. Closing this issue since I don't have the bandwidth to work on it right now.

Feel free to pick up my branch.

**Pull request closed without merging**